tests/test_pb_well_models.py

```diff
@@ -0,0 +1,198 @@
+from npg_id_generation.pac_bio import PacBioEntity
+from sqlalchemy.orm import Session
+
+from lang_qc.db.helper.qc import get_qc_states_by_id_product_list
+from lang_qc.db.helper.wells import WellWh
+from lang_qc.db.mlwh_schema import PacBioRunWellMetrics
+from lang_qc.models.pacbio.well import PacBioWellFull, PacBioWellSummary
+from tests.conftest import compare_dates, insert_from_yaml
+from tests.fixtures.well_data import load_data4well_retrieval, load_dicts_and_users
+
+yaml_is_loaded: bool = False
+
+
+def _prepare_data(
+    mlwhdb_session: Session,
+    qcdb_session: Session,
+    run_name: str,
+    well_label: str,
+    plate_number: int = None,
+):
+    """Loads LIMS data for one well.
+
+    Returns a tuple of an mlwh db row and QC state model for that well.
+    """
+
+    global yaml_is_loaded
+
+    if yaml_is_loaded is False:
```

Comment:
Why does this not want to be a fixture? Then you wouldn't need to do your own caching.